tests/test_e2e.py

```diff
@@ -61,6 +62,35 @@ def test_kmeans_summary():
     validate_run(run, log_charts=True)
 
 
+@pytest.mark.filterwarnings("error::neptune.common.warnings.NeptuneUnsupportedType")
+def test_unsupported_object():
+    """This method checks if Neptune throws a `NeptuneUnsupportedType` warning if expected metadata
+    is not found or skips trying to log such metadata"""
+
+    from sklearn.model_selection import GridSearchCV
+
+    run = init_run()
```

Comment:
Don't we need to stop the run somewhere? How about using a context manager
`with init_run() as run:` ?

Reply:
It's running as a script, so the run would be stopped automatically once the script terminates?
I followed the same design as the existing tests, except not using the `validate_run()` function

Reply:
Yes, but I guess that the design in this module in general isn't very good. That could be a separate task to refactor it. If the script terminates the run will (eventually) terminate due to garbage collection, but what if an exception happens somewhere along the way? Let's refactor it here and I'll add a task for myself to adjust the rest of the module's content, ok?